### PR TITLE
Add testbench for easier testing

### PR DESCRIPTION
**Pull request by hator**
This pull request:
1. Splits cachehog.c into two files: 
  * cachehog_lib.h -- reusable functions for using cachehog transmission
  * cachehog.c -- main program as previously
2. Adds testbench program

Testbench automatically forks itself into transmitter and receiver, tries to send the message and calculates error rates.

This is preparation for work on error correcting codes.

### CODE REVIEW

File: testbench.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+void transmitter(const uint8_t* message, size_t size) {
+  init_cachehog();
+
+  for(int i = 0; i < size; i++) {
+    transmit_word(message[i]);
+  }
+}
+
+void after_receiving();
```
Comment (AwesomePatrol) [inline testbench.c:23]: CS: Move after global variable declaration

Preferably declare all in a separate header file.

File: testbench.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+void recv_sighandler(int sig) {
+    after_receiving();
```
Comment (AwesomePatrol) [inline testbench.c:26]: CS: inconsistency. indentation here: 4 spaces, later 2.

File: README.md [hunk 3]
```diff
@@ -81,6 +81,33 @@ Not bad.
...
+    $ ./testbench
+    Message size: 28
+    Estimated running time: 22 sec
+    Received 28 bytes (224 bits)
+    Lost 0 bytes
+    ORG: Lorem ipsum dolor sit amet.
+    REC: Lorm ipsum dolorsit amet.
+    Total error bits: 10, bit error rate: 0.044643
+
+NOTE: testbench does not run any diffing algorithm so if any of byte is lost in
```
Comment (AwesomePatrol) [inline README.md:104]: `s/byte/bytes/`

File: testbench.c [hunk 4]
```diff
@@ -7,7 +7,7 @@
 #include <signal.h>
 #include "cachehog_lib.h"
 
-uint8_t* message;
+const uint8_t* message;
```
Comment (AwesomePatrol) [inline testbench.c]: I think this change belongs to the previous commit.

File: testbench.c [hunk 5]
```diff
@@ -52,6 +52,28 @@ int calc_error_count() {
...
+const char* byte_to_binary(int x) {
+  static char b[9];
+  b[0] = '\0';
+
+  for(int i = 0; i < 8; i++) {
+    b[i] = (x & (128 >> i)) ? '1' : '0';
+  }
+
+  return b;
+}
```
Comment (AwesomePatrol) [inline testbench.c:64]: CS: separate function implementation with a single newline.

**Comment by zyla** [comment]
Awesome stuff, thanks.

**Pull request merged**